src/expr/src/agg/def.rs

```diff
@@ -410,6 +408,10 @@ pub mod agg_kinds {
                 | AggKind::PercentileCont
                 | AggKind::PercentileDisc
                 | AggKind::Mode
+                // FIXME(wrj): move `BoolAnd` and `BoolOr` out
+                //  after we support general merge in stateless_simple_agg
+                | AggKind::BoolAnd
+                | AggKind::BoolOr
```

Comment:
Here it comes the problem. For updatable `bool_and` function, its total function is `sum`. But the append-only version is `bool_and` itself. Since we don't record `append_only` in `AggCall`, it's hard to determine the total function in `partial_to_total`.

Actually in my plan, we will no longer have the `partial_to_total` in the future. Instead, each aggregate function will have an optional `merge` function and we will call `merge` on the original function in `StatelessSimpleAgg`. This will be done in the future PR if it sounds good.

Reply:
Seems we can rewrite retractable `bool_or(x)` to `count(x) filter (where x) > 0`, `bool_and(x)` to `count(x) filter (where not x) = 0`, then they can be 2-phased automatically. Also, I think `bool_and(bool_and(x))` should automatically work too.

IMO `partial_to_total` should only care about the input and output of partial and total phase. Whether it's `append only` or not doesn't matter. `append only` affects the implementation, while implementation is orthogonal to the ability to be 2-phased.

Reply:
It's proven that 2-phase agg can't work on `bool_and` now. Because the total phase aggregates on the result (all true?) rather than the state (false count), some information may loss.